KCP CLI: rename to kcp, documentation (#270)
kyma-project/control-plane: docs/cli/commands/kcp_upgrade_kyma.md

# kcp upgrade kyma
Upgrades or reconfigures Kyma on one or more Kyma Runtimes.

## Synopsis

Upgrades or reconfigures Kyma on targets of Runtimes.
The upgrade is performed by Kyma Control Plane (KCP) within a new orchestration asynchronously. The ID of the orchestration is returned by the command upon success.
The targets of Runtimes are specified via the `--target` and `--target-exclude` options. At least one `--target` must be specified.

> The Kyma version and configurations to use for the upgrade are taken from the Kyma Control Plane during the processing of the orchestration.

Reviewer: ```suggestion
The Kyma version and configurations to use for the upgrade are taken from Kyma Control Plane during the processing of the orchestration.
```

      --strategy string              Orchestration strategy to use. Currently the only supported strategy is parallel. (default "parallel")
  -t, --target stringArray           List of Runtime target specifiers to include. You can specify this option multiple times.
                                     A target specifier is a comma-separated list of the following selectors:
                                       all                 : All Runtimes provisioned successfully and not deprovisioning
                                       account=<REGEXP>    : Regex pattern to match against the Runtime's global account field, e.g. "CA50125541TID000000000741207136", "CA.*"
                                       subaccount=<REGEXP> : Regex pattern to match against the Runtime's subaccount field, e.g. "0d20e315-d0b4-48a2-9512-49bc8eb03cd1"
                                       region=<REGEXP>     : Regex pattern to match against the Runtime's provider region field, e.g. "europe|eu-"
                                       runtime-id=<ID>     : Runtime ID is used to indicate a specific Runtime

>   -e, --target-exclude stringArray   List of runtime target specifiers to exclude. You can specify this option multiple times.

Reviewer: ```suggestion
  -e, --target-exclude stringArray   List of Runtime target specifiers to exclude. You can specify this option multiple times.
```

>                                      A target specifier is a comma-separated list of the selectors described under --target option.

Reviewer: ```suggestion
                                     A target specifier is a comma-separated list of the selectors described under the --target option.
```

---
title: kcp upgrade kyma
---

> Upgrade or reconfigure Kyma on one or more Kyma runtimes

Reviewer: ```suggestion
Upgrades or reconfigures Kyma on one or more Kyma Runtimes.
```

## Synopsis

Upgrades or reconfigures Kyma on targets of runtimes.

> The upgrade is performed by the Kyma Control plane within a new orchestration asynchronously, the id of which is returned by the command upon success.

Reviewer: ```suggestion
The upgrade is performed by Kyma Control Plane (KCP) within a new orchestration asynchronously. The ID of the orchestration is returned by the command upon success.
```

> The targets of runtimes are specified via the --target and --target-exclude options. At lease one --target must be specified.

Reviewer: ```suggestion
The targets of Runtimes are specified using the `--target` and `--target-exclude` options. At least one `--target` must be specified.
```

> The Kyma version and configurations to use for the upgrade is taken from the Kyma Control Plane during processing of the orchestration.

Reviewer: ```suggestion
The Kyma version and configurations to use for the upgrade are taken from Kyma Control Plane during the processing of the orchestration.
```

```bash

> kcp upgrade kyma --target <TARGET SPEC> ... [--target-exclude <TARGET SPEC> ...] [flags]

Reviewer: ```suggestion
kcp upgrade kyma --target {TARGET SPEC} ... [--target-exclude {TARGET SPEC} ...] [flags]
```

```

## Examples

```
  kcp upgrade kyma --target all --schedule maintenancewindow     Upgrade Kyma on all runtimes in their next respective maintenance window hours
  kcp upgrade kyma --target "account=CA.*"                       Upgrade Kyma on runtimes of all Global Accounts starting with CA
  kcp upgrade kyma --target all --target-exclude "account=CA.*"  Upgrade Kyma on runtimes of all Global Accounts not starting with CA
  kcp upgrade kyma --target "region=europe|eu|uk"                Upgrade Kyma on runtimes whose region belongs to Europe
```

## Options

```

>       --dry-run                      Perform the orchestration without executing the actual upgrage operations for the runtimes. The details can be obtained using the "kcp orchestrations" command

Reviewer: Please apply comments from the previous document's table here as well.
